### ISSUE TITLE
[Action required] Deprecation: Removal of Sphinx context injection at build time

### ISSUE THREAD

**Issue by jhamman** [pydata/xarray#9251]
### What is your issue?

I got this deprecation notice for Xarray today. Seems like a patch is going to be needed to our docs setup.
 
> Hello,
> We are announcing the deprecation of Sphinx context injection at build time for all the projects. The deprecation date is set on Monday, October 7th, 2024. After this date, Read the Docs won't install the readthedocs-sphinx-ext extension and won't manipulate the project's conf.py file.
> 
> This will get us closer to our goal of having all projects build on Read the Docs be the exact same as on other build environments, making understanding of documentation builds much easier to understand.
> 
> You can read our [blog post on the deprecation](https://about.readthedocs.com/blog/2024/07/addons-by-default/) for all the information about possible impacts of this change, in particular the READTHEDOCS and other variables in the Sphinx context are no longer set automatically.
> 
> Deprecation timeline
> We understand this change will affect thousands of projects, so we have a timeline to communicate this deprecation to our users effectively.
> 
> July 29, 2024: disable Sphinx context injection and enable Read the Docs Addons for all new projects created.
> October 7, 2024: disable Sphinx context injection and enable Read the Docs Addons for all projects.
> Are my projects affected?
> You have at least one project with a canonical custom domain defined. After the deprecation date, Read the Docs will stop defining [html_baseurl](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_baseurl) Sphinx configuration, which means that you will need to define it by yourself to keep the canonical custom domain properly configured.
> 
> We found the following projects where you are a maintainer are impacted by this change:
> 
> [xray](https://readthedocs.org/projects/xray/)
> 
> How I can update my projects?
> You can use the following Python snippet as a guide to update your conf.py file. The snippet uses the [READTHEDOCS_CANONICAL_URL](https://docs.readthedocs.io/page/reference/environment-variables.html#envvar-READTHEDOCS_CANONICAL_URL) environment variable to define [html_baseurl](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_baseurl) and injects [READTHEDOCS](https://docs.readthedocs.io/page/reference/environment-variables.html#envvar-READTHEDOCS) variable into the [html_context](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_context) to keep the previous behavior:
> 
```python
import os

# Set canonical URL from the Read the Docs Domain
html_baseurl = os.environ.get("READTHEDOCS_CANONICAL_URL", "")

# Tell Jinja2 templates the build is running on Read the Docs
if os.environ.get("READTHEDOCS", "") == "True":
    html_context["READTHEDOCS"] = True
```
> Closing notes
> We recommend you to read the full announcement [Read the Docs Addons enabled by default](https://about.readthedocs.com/blog/2024/07/addons-by-default/) in our blog that contains a more detailed explanation and other similar changes that may affect other projects you maintain as well.
> 
> [Contact us](https://readthedocs.org/support/) or reply to this email if you have any questions or feedback about this change.
> 
> Keep documenting,
> Read the Docs

**Comment by VeckoTheGecko** [comment]
This deprecation landed on October 7 2024 and doesn't seem to have affected us at all. We don't use the `html_baseurl` in `conf.py` nor do we use the `html_context["READTHEDOCS"]` key.

I think let's just close - there's no need to implement the suggested fix as nothing broke

**Comment by keewis** [comment]
yeah, I agree, looks like this didn't affect us. Thanks for the triaging!
